app/views/components/_markdown-editor.html.erb

```diff
@@ -0,0 +1,44 @@
+<%
+  label_text ||= "Enter your markdown here"
+  textarea ||= {}
+  component_classes = %w[app-c-markdown-editor govuk-form-group]
+  textarea[:id] = textarea[:id].present? ? textarea[:id] : textarea[:name]
+%>
+<%= tag.div class: component_classes do %>
+
+  <%= render "govuk_publishing_components/components/label", {
+    html_for: textarea[:id].present? ? textarea[:id] : textarea[:name],
+    text: label_text
+  } %>
+
+   <div class="app-c-markdown-editor__container js-markdown-editor__container">
+    <div class="app-c-markdown-editor__head js-markdown-editor__head">
```

Comment:
Should the JS classes only be added with JS, rather than being in the DOM by default? (`js-markdown-editor__head` etc).

Reply:
We're using the `js-` prefixed classes to query elements from the DOM - sort of a way of separating style from function. (e.g. this class is used for querying elements, not styling them)

Reply:
Thanks for clarifying - no problem here then!